### PR TITLE
add symlink variant (needed for AFS)

### PR DESCRIPTION
**Pull request by healther**
This PR adds a symlinks variant to ncurses. This is needed for installs on symlinked filesystems (e.g. AFS)

### CODE REVIEW

**Comment by citibeth** [comment]
This PR was easy to approve because it hooks up an upstream option to a
variant, and doesn't change anything for existing users.  I would lean
toward leaving the deafult the same as it is for the upstream package.

File: var/spack/repos/builtin/packages/ncurses/package.py [hunk 1]
```diff
@@ -42,6 +42,9 @@ class Ncurses(Package):
 
     patch('patch_gcc_5.txt', when='%gcc@5.0:')
 
+    variant('symlinks', default=False,
+        description='Enables symlinks. Needed on AFS filesystem.')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ncurses/package.py]: This line needs to be indented by 4 more spaces in order to pass the flake8 tests.

**Comment by tgamblin** [comment]
@adamjstewart: I resolved conflicts on this PR to match your prior ncurses change.  Can you check this out and see if it still works for you?

**Comment by healther** [comment]
LGTM

File: var/spack/repos/builtin/packages/ncurses/package.py [hunk 2]
```diff
@@ -54,3 +57,5 @@ def configure_args(self):
             '--enable-pc-files',
             '--with-pkg-config-libdir={0}/lib/pkgconfig'.format(self.prefix)
         ]
+        if '+symlinks' in self.spec:
+            opts += ["--enable-symlinks"]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ncurses/package.py]: `opts` is never returned. I doubt this package actually works. Did you test this at all?

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ncurses/package.py]: It looks like this happened during @tgamblin's develop merge. Can you add `return opts` to the end?

Comment (healther) [inline var/spack/repos/builtin/packages/ncurses/package.py]: You are right. The return was missing and the ```configure(*opts)``` was red.

 I really shouldn't comment without explicit testing... Or while being hungry. Sorry for that

**Pull request merged**